syntax = "proto3";

import "spiderrock_common.proto";


package spiderrock.protobuf;


/*
    This table contains live option quote records from OPRA (equities) or the listing exchange (futures).  Each record contains up to two price levels and represents a live snapshot of the book for a specific option series.  There are typically 1mm+ records in this table if all ticker sources are enabled.
*/    
message OptionNbboQuote {
    message PKey {
        
        OptionKey okey = 10;
    }


    MessageMetadata _meta = 1;
    
    PKey pkey = 2;
    UpdateType update_type = 100;
    float bid_price = 103; // bid price
    float ask_price = 106; // ask price
    int32 bid_size = 109; // bid size in contracts (largest exch quote)
    int32 ask_size = 112; // ask size in contracts (largest exch quote)
    int32 cum_bid_size = 115; // bid size in contracts (total nbbo size)
    int32 cum_ask_size = 118; // ask size in contracts (total nbbo size)
    OptExch bid_exch = 121; // first (or largest remaining) exchange at bid price
    OptExch ask_exch = 124; // first (or largest remaining) exchange at ask price
    int64 bid_mask = 161; // exchange bid bit mask
    int64 ask_mask = 162; // exchange ask bit mask
    OptMktType bid_mkt_type = 133; // bid side quote flags (if any)
    OptMktType ask_mkt_type = 136; // ask side quote flags (if any)
    float bid_price2 = 139; // 2nd best bid price
    float ask_price2 = 142; // 2nd best ask price
    int32 cum_bid_size2 = 145; // cumulative size at 2nd price
    int32 cum_ask_size2 = 148; // cumulative size at 2nd price
    int32 bid_time = 151; // last bid price change (milliseconds since midnight) calculated from the srcTimestamp
    int32 ask_time = 154; // last ask price change (milliseconds since midnight) calculated from the srcTimestamp
    int64 src_timestamp = 157; // source high precision timestamp (if available)
    int64 net_timestamp = 160; // inbound packet PTP timestamp from SR gateway switch;usually syncronized with facility grandfather clock
}


